lib/util/expected-state.js

```diff
@@ -213,11 +222,15 @@ module.exports = ExpectedState = {
         this._components.clear();
     },
     clearDynamicArea(dynamicArea) {
-        let components = this._dynamicAreas.get(dynamicArea);
-        if (components) {
-            for (let component of components) {
-                this.delete(component);
-            }
+        if (dynamicArea) {
+            let components = Array.from(this._dynamicAreas.get(dynamicArea));
```

Comment:
need to do a check that the dynamicArea.get returns something before you call Array.from or else it will explode. 

Can do if (dynamicArea && this._dynamicAreas.has(dynamicArea) on line 225

Also you dont need to do Array.from, you can iterate over the set directly, which means you would just do if (components) instead of that other checky stuff